CRM/MembershipExtras/Job/OfflineAutoRenewal/PaymentSchemePlan.php

```diff
@@ -327,4 +329,16 @@ protected function calculateRecurringContributionTotalAmount($recurringContribut
     return MoneyUtilities::roundToCurrencyPrecision($totalAmount);
   }
 
+  private function dispatchPaymentSchemeRenewalHook() {
+    $nullObject = CRM_Utils_Hook::$_nullObject;
+    CRM_Utils_Hook::singleton()->invoke(
+      ['paymentProcessorId', 'recurContributionId', 'paymentPlanSchedule'],
+      $this->newRecurringContribution['payment_processor_id'],
+      $this->newRecurringContributionID,
+      $this->paymentPlanSchedule,
```

Comment:
@omarabuhussein  I don't think GoCardless care about payment plane schedule, what it needs is the list of contribution ID, amount, current that we want to create payments in GoCardless, which I think they were created in the part of renewal.

Reply:
hmmm... I am not following, because that is not what was discussed and agreed on, wasn't the point of this ticket/PR https://github.com/compucorp/uk.co.compucorp.membershipextras/pull/463 to create the API to send the data in specific format so that you can use it in GoCardless to create the payment schedule on GoCardless which also follows the format GoCardless expected ?

Reply:
Update 1: I had a chat with Ererawat, and it seems that the payment schedule was needed before because we wanted not to create the contributions upfront where we create a payment schedule on GoCardless instead, but given this changed now to creating the contributions upfront, passing only the payment schedule here has no value, so I've changed the hook above to pass the contributions (instalments) list after autorenewal.